src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -5206,16 +5206,18 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           case MethodType(p :: _, _)   => p.isImplicit            // implicit method requires no args
           case _                       => true                    // catches all others including NullaryMethodType
         }
-        def isPlausible(m: Symbol) = m.alternatives exists (m => requiresNoArgs(m.info))
+        def isPlausible(m: Symbol) = !m.isPackage && m.alternatives.exists(x => requiresNoArgs(x.info))
 
         def maybeWarn(s: String): Unit = {
           def warn(message: String)         = context.warning(lit.pos, s"possible missing interpolator: $message")
           def suspiciousSym(name: TermName) = context.lookupSymbol(name, _ => true).symbol
-          def suspiciousExpr                = InterpolatorCodeRegex findFirstIn s
+          val suspiciousExpr                = InterpolatorCodeRegex findFirstMatchIn s
           def suspiciousIdents              = InterpolatorIdentRegex findAllIn s map (s => suspiciousSym(TermName(s drop 1)))
 
           if (suspiciousExpr.nonEmpty)
-            warn("detected an interpolated expression") // "${...}"
+            suspiciousExpr filter (!_.group(1).trim.isEmpty) foreach (_ =>
+              warn("detected an interpolated expression") // "${...}"
+            )
```

Comment:
I thought you were going to apply the `suspiciousSym` test if the text is an identifier. Then `${junk}` gets a pass like `$pass`. That might solve most conflicts with templating.